File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/db/documents/DocumentsPersistanceDBModule.java
package com.fintechplatform.ui.profile.db.documents;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DocumentsPersistanceDBModule {

    @Provides
    @Singleton
    DocumentsPersistanceDB providesDocumentsPersistanceDB(DocumentsDB documentsDB) {
        return new DocumentsPersistanceDB(documentsDB);
    }

    @Provides
    @Singleton
    DocumentsDB providesDocumentsDB() {
        return new DocumentsDB();
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/db/documents/DocumentsPersistanceDBModule.kt
package com.fintechplatform.ui.profile.db.documents

import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class DocumentsPersistanceDBModule {

    @Provides
    @Singleton
    internal fun providesDocumentsPersistanceDB(documentsDB: DocumentsDB): DocumentsPersistanceDB = DocumentsPersistanceDB(documentsDB)

    @Provides
    @Singleton
    internal fun providesDocumentsDB(): DocumentsDB = DocumentsDB()
}